pygatt/backends/gatttool/gatttool.py

```diff
@@ -316,27 +404,24 @@ def char_write_handle(self, handle, value, wait_for_response=False):
         :param value:
         :param wait_for_response:
         """
-        with self._connection_lock:
-            hexstring = ''.join('%02x' % byte for byte in value)
-
-            if wait_for_response:
-                cmd = 'req'
-            else:
-                cmd = 'cmd'
-
-            cmd = 'char-write-%s 0x%02x %s' % (cmd, handle, hexstring)
-
-            log.debug('Sending cmd=%s', cmd)
-            self._con.sendline(cmd)
-
-            if wait_for_response:
-                try:
-                    self._expect('Characteristic value written successfully')
-                except NoResponseError:
```

Comment:
Nothing in pygatt raises this exception; seems like dead code. GATTToolBackend._expect would have previously raised NotificationTimeout error only.

Looks like a bug that no longer need fixing. I'll remove this Exception from pygatt entirely.